src/password/scripts/directives.js

```diff
@@ -90,11 +90,12 @@
                         .then(function (response) {
                             scope.fields = response.data.actions.PUT;
                         }, function (response) {
-                            // If the response returns a 404 and we have a token.
-                            if (response.status === 404 && angular.isDefined(TOKEN)) {
-                                // There was a problem with the token.
+                            if (angular.isDefined(TOKEN)) {
                                 scope.tokenError = true;
                             }
+                            if (angular.isDefined(response.data) && angular.isDefined(response.data.detail)) {
+                                scope.errorMessage = response.data.detail;
```

Comment:
It would be nice to know the other possible error keys.

For the moment, combining it with `scope.errors` and looping over the response in a similar manner means, in the least, `non_field_errors` is accounted for.